lib/manager.js

```diff
@@ -91,11 +91,13 @@ export async function get(deviceType, versionOrRange) {
 	}
 	const version = await utils.resolveVersion(deviceType, versionOrRange);
 	const isFresh = await cache.isImageFresh(deviceType, version);
-	if (isFresh) {
-		return cache.getImage(deviceType, version);
-	} else {
-		return doDownload(deviceType, version);
-	}
+	const $stream = isFresh
+		? await cache.getImage(deviceType, version)
+		: await doDownload(deviceType, version);
+	// schedule the 'version' event for the next iteration of the event loop
+	// so that callers have a chance of adding an event handler
+	setImmediate(() => $stream.emit('version', version));
+	return $stream;
```

Comment:
Emitting this new `version` event allows `balena os download` to print the OS version actually being downloaded (the resolved version) when the user requests something like:
* `>2021.01.1 <2021.04.1`
* `'recommended'`
* `'latest'`
* `'default'`

I understand that emitting a new event is **not** a breaking change, but let me know if you don't think this is the case.

**Also important: I am assuming that `balena-image-manager` is only used with Node.js, not in a web browser (e.g. web dashboard).** Is this right? There is a `cache.js` file that seems to deal with files in a filesystem and I assume it doesn't work in a web browser.

Reply:
You are right, afaik balena-image-manager is only used by the CLI & in node.

Reply:
Maybe we could namespace it just to be super sure (eg: `balena-image-manager:resolved-version`), but yes that's not a breaking change in any way.

Reply:
Done 👍